gwcli/gateway.py

```diff
@@ -364,22 +372,59 @@ def summary(self):
 
 class Target(UIGroup):
 
-    help_info = '''
-                The iscsi target is the name that the group of gateways are 
-                known as by the iscsi initiators (clients).
-                '''
-
+    help_intro = '''
+                 The iscsi target is the name that the group of gateways are
+                 known as by the iscsi initiators (clients).
+                 '''
     def __init__(self, target_iqn, parent):
-
         UIGroup.__init__(self, target_iqn, parent)
         self.target_iqn = target_iqn
+
+        self.controls = {}
+
         self.gateway_group = GatewayGroup(self)
         self.client_group = Clients(self)
         self.host_groups = HostGroups(self)
 
+    def _get_controls(self):
+        return self._controls.copy()
+
+    def _set_controls(self, controls):
+        self._controls = controls.copy()
+        self._refresh_control_values()
+
+    controls = property(_get_controls, _set_controls)
+
     def summary(self):
         return "Gateways: {}".format(len(self.gateway_group.children)), None
 
+    def ui_command_info(self):
+        self.logger.debug("CMD: info")
+
```

Comment:
Should the Target class be UINode instead of UIGroup, so we can use the info helpers? We can then drop all this and just add a display_attributes arrary with target_iqn and control_values and this function is just a call to

console_message(get_info())

?

I honestly did not understand the difference between uigroup and uinode other than the info difference, so no idea if that is ok wrt python/object oriented design. It just looked easier code wise.

My last review comment. Sorry for the sporadic review comments over multiple days :)

Reply:
From my quick glance, I thought the ```UIGroup``` is allowed to have children and the ```UINode``` was designed to be a leaf. I might be wrong, though.

Reply:
... never mind, they seem to basically be the same thing. updated.